Build:

$ make -f makefile.bsd -C build

Run:

$ qemu -fda build/grub.ok -tftp build

diff -Nurp os.orig/build/makefile.bsd os/build/makefile.bsd
--- os.orig/build/makefile.bsd	2004-12-05 16:01:38.000000000 +0100
+++ os/build/makefile.bsd	2009-03-06 14:51:21.000000000 +0100
@@ -19,8 +19,8 @@ OBJS	= ../dev/kbd.o ../dev/vga.o ../dev/
 
 .SUFFIXES: .o .c .s
 .c.o .s.o: .SILENT
-	echo Compiling $(.IMPSRC:T)
-	$(CC) $(CFLAGS) -c $(.IMPSRC) -o $(.IMPSRC:R).o
+	echo Compiling $<
+	$(CC) $(CFLAGS) -c $< -o $@
 
 # targets
 
diff -Nurp os.orig/machdep/l.s os/machdep/l.s
--- os.orig/machdep/l.s	2004-12-05 16:38:42.000000000 +0100
+++ os/machdep/l.s	2009-03-06 14:54:27.000000000 +0100
@@ -35,7 +35,7 @@
 		iret
 .endm
 
-.macro	irq a,
+.macro	irq a
 	.globl	irq\a
 	irq\a:
 		pusha
